src/main/scala/io/iohk/ethereum/consensus/Consensus.scala

```diff
@@ -1,41 +1,187 @@
 package io.iohk.ethereum.consensus
 
-import io.iohk.ethereum.domain.Block
-import io.iohk.ethereum.ledger.BlockExecutionError.ValidationBeforeExecError
-import io.iohk.ethereum.ledger.BlockExecutionSuccess
+import akka.util.ByteString
+import io.iohk.ethereum.consensus.blocks.{BlockGenerator, TestBlockGenerator}
+import io.iohk.ethereum.consensus.validators.Validators
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.ledger.BlockExecutionError.{ValidationAfterExecError, ValidationBeforeExecError}
+import io.iohk.ethereum.ledger.Ledger.VMImpl
+import io.iohk.ethereum.ledger.{BlockExecutionError, BlockExecutionSuccess, BlockPreparator}
 import io.iohk.ethereum.nodebuilder.Node
-import io.iohk.ethereum.validators.BlockHeaderValidator
+import io.iohk.ethereum.utils.{BlockchainConfig, Logger}
+import org.spongycastle.util.encoders.Hex
 
 /**
- * Abstraction for a consensus protocol.
+ * Abstraction for a consensus protocol implementation.
  *
  * @see [[io.iohk.ethereum.consensus.Protocol Protocol]]
  */
-// FIXME Lot's of stuff to do...
 trait Consensus {
+  /**
+   * The type of configuration [[io.iohk.ethereum.consensus.FullConsensusConfig#specific specific]]
+   * to this consensus protocol implementation.
+   */
   type Config <: AnyRef /*Product*/
 
   def protocol: Protocol
 
   def config: FullConsensusConfig[Config]
 
-  /** Returns `true` if this is the standard Ethereum PoW consensus protocol (`ethash`). */
-  final def isEthash: Boolean = protocol.isEthash
+  /**
+   * This is the VM used to prepare and generate blocks.
```

Comment:
VM does not prepare nor generate blocks :)

Reply:
It is _used_ in the process

Reply:
I'm thinking about changing the wording to `VM used while preparing and generating blocks` in order to reduce confusion.

Reply:
[Changed](https://github.com/input-output-hk/mantis/pull/410/commits/3a855d3dd91e8d728692544a09d46fdf45a7d85b)